Compilation errors

# Basic information
  - Type of the CPU. Intel

# Compile issues
  - Which OS do you use? Linux
  ```
cmake -DCUDA_ENABLE=OFF -DMICROHTTPD_ENABLE=OFF  -DHWLOC_ENABLE=OFF -DOpenCL_ENABLE=OFF ..

Scanning dependencies of target xmr-stak-c                                                                                           [26/7897]
[  7%] Building C object CMakeFiles/xmr-stak-c.dir/backend/cpu/crypto/c_groestl.c.o
[  7%] Building C object CMakeFiles/xmr-stak-c.dir/backend/cpu/crypto/c_blake256.c.o                                  
[ 11%] Building C object CMakeFiles/xmr-stak-c.dir/backend/cpu/crypto/c_jh.c.o
[ 15%] Building C object CMakeFiles/xmr-stak-c.dir/backend/cpu/crypto/c_skein.c.o
[ 19%] Building C object CMakeFiles/xmr-stak-c.dir/backend/cpu/crypto/c_keccak.c.o
[ 23%] Linking C static library bin/libxmr-stak-c.a
[ 23%] Built target xmr-stak-c
Scanning dependencies of target xmr-stak-backend
[ 26%] Building CXX object CMakeFiles/xmr-stak-backend.dir/jconf.cpp.o
[ 30%] Building CXX object CMakeFiles/xmr-stak-backend.dir/version.cpp.o
[ 34%] Building CXX object CMakeFiles/xmr-stak-backend.dir/backend/cpu/hwlocMemory.cpp.o
[ 38%] Building CXX object CMakeFiles/xmr-stak-backend.dir/backend/cpu/jconf.cpp.o
[ 42%] Building CXX object CMakeFiles/xmr-stak-backend.dir/backend/cpu/minethd.cpp.o
[ 46%] Building CXX object CMakeFiles/xmr-stak-backend.dir/backend/backendConnector.cpp.o
[ 50%] Building CXX object CMakeFiles/xmr-stak-backend.dir/backend/globalStates.cpp.o
[ 53%] Building CXX object CMakeFiles/xmr-stak-backend.dir/backend/cpu/crypto/cryptonight_common.cpp.o
[ 57%] Building CXX object CMakeFiles/xmr-stak-backend.dir/http/httpd.cpp.o
[ 61%] Building CXX object CMakeFiles/xmr-stak-backend.dir/http/webdesign.cpp.o
[ 65%] Building CXX object CMakeFiles/xmr-stak-backend.dir/misc/console.cpp.o
[ 69%] Building CXX object CMakeFiles/xmr-stak-backend.dir/misc/executor.cpp.o
In file included from /home/dlh/git/xmr-stak/xmrstak/backend/cpu/minethd.cpp:43:0:
/home/dlh/git/xmr-stak/xmrstak/backend/cpu/autoAdjust.hpp: In member function ‘bool xmrstak::cpu::autoAdjust::printConfig()’:
/home/dlh/git/xmr-stak/xmrstak/backend/cpu/autoAdjust.hpp:41:23: error: ‘class jconf’ has no member named ‘IsCurrencyMonero’
   if(::jconf::inst()->IsCurrencyMonero())
                       ^~~~~~~~~~~~~~~~
/home/dlh/git/xmr-stak/xmrstak/backend/cpu/autoAdjust.hpp:43:20: error: ‘MONERO_MEMORY’ was not declared in this scope                [0/7897]
    hashMemSizeKB = MONERO_MEMORY / 1024u;
                    ^~~~~~~~~~~~~
[ 73%] Building CXX object CMakeFiles/xmr-stak-backend.dir/misc/telemetry.cpp.o
/home/dlh/git/xmr-stak/xmrstak/backend/cpu/autoAdjust.hpp:48:20: error: ‘AEON_MEMORY’ was not declared in this scope
    hashMemSizeKB = AEON_MEMORY / 1024u;
                    ^~~~~~~~~~~
[ 76%] Building CXX object CMakeFiles/xmr-stak-backend.dir/misc/uac.cpp.o
[ 80%] Building CXX object CMakeFiles/xmr-stak-backend.dir/misc/utility.cpp.o
[ 84%] Building CXX object CMakeFiles/xmr-stak-backend.dir/net/jpsock.cpp.o
[ 88%] Building CXX object CMakeFiles/xmr-stak-backend.dir/net/socket.cpp.o
make[2]: *** [CMakeFiles/xmr-stak-backend.dir/build.make:159: CMakeFiles/xmr-stak-backend.dir/backend/cpu/minethd.cpp.o] Błąd 1
make[2]: *** Oczekiwanie na niezakończone zadania....
make[1]: *** [CMakeFiles/Makefile2:105: CMakeFiles/xmr-stak-backend.dir/all] Błąd 2
make: *** [Makefile:130: all] Błąd 2
  ```
  ```
-- Configuring done                 
-- Generating done                       
-- Build files have been written to: /home/dlh/git/xmr-stak/xmrstak
-- Cache values                              
CMAKE_AR:FILEPATH=/usr/bin/ar         
CMAKE_BUILD_TYPE:STRING=Release                         
CMAKE_COLOR_MAKEFILE:BOOL=ON     
CMAKE_CXX_COMPILER:FILEPATH=/usr/bin/c++
CMAKE_CXX_COMPILER_AR:FILEPATH=/usr/bin/gcc-ar
CMAKE_CXX_COMPILER_RANLIB:FILEPATH=/usr/bin/gcc-ranlib
CMAKE_CXX_FLAGS:STRING=                
CMAKE_CXX_FLAGS_DEBUG:STRING=-g             
CMAKE_CXX_FLAGS_MINSIZEREL:STRING=-Os -DNDEBUG
CMAKE_CXX_FLAGS_RELEASE:STRING=-O3 -DNDEBUG     
CMAKE_CXX_FLAGS_RELWITHDEBINFO:STRING=-O2 -g -DNDEBUG
CMAKE_C_COMPILER:FILEPATH=/usr/bin/cc  
CMAKE_C_COMPILER_AR:FILEPATH=/usr/bin/gcc-ar
CMAKE_C_COMPILER_RANLIB:FILEPATH=/usr/bin/gcc-ranlib
CMAKE_C_FLAGS:STRING=            
CMAKE_C_FLAGS_DEBUG:STRING=-g
CMAKE_C_FLAGS_MINSIZEREL:STRING=-Os -DNDEBUG                                                                                         [22/8659]
CMAKE_C_FLAGS_RELEASE:STRING=-O3 -DNDEBUG
CMAKE_C_FLAGS_RELWITHDEBINFO:STRING=-O2 -g -DNDEBUG
CMAKE_EXE_LINKER_FLAGS:STRING=
CMAKE_EXE_LINKER_FLAGS_DEBUG:STRING=
CMAKE_EXE_LINKER_FLAGS_MINSIZEREL:STRING=
CMAKE_EXE_LINKER_FLAGS_RELEASE:STRING=
CMAKE_EXE_LINKER_FLAGS_RELWITHDEBINFO:STRING=
CMAKE_EXPORT_COMPILE_COMMANDS:BOOL=OFF
CMAKE_INSTALL_PREFIX:PATH=/home/dlh/git/xmr-stak/xmrstak
CMAKE_LINKER:FILEPATH=/usr/bin/ld
CMAKE_LINK_STATIC:BOOL=OFF
CMAKE_MAKE_PROGRAM:FILEPATH=/usr/bin/make
CMAKE_MODULE_LINKER_FLAGS:STRING=
CMAKE_MODULE_LINKER_FLAGS_DEBUG:STRING=
CMAKE_MODULE_LINKER_FLAGS_MINSIZEREL:STRING=
CMAKE_MODULE_LINKER_FLAGS_RELEASE:STRING=
CMAKE_MODULE_LINKER_FLAGS_RELWITHDEBINFO:STRING=
CMAKE_NM:FILEPATH=/usr/bin/nm
CMAKE_OBJCOPY:FILEPATH=/usr/bin/objcopy
CMAKE_OBJDUMP:FILEPATH=/usr/bin/objdump
CMAKE_RANLIB:FILEPATH=/usr/bin/ranlib
CMAKE_SHARED_LINKER_FLAGS:STRING=
CMAKE_SHARED_LINKER_FLAGS_DEBUG:STRING=
CMAKE_SHARED_LINKER_FLAGS_MINSIZEREL:STRING=
CMAKE_SHARED_LINKER_FLAGS_RELEASE:STRING=
CMAKE_SHARED_LINKER_FLAGS_RELWITHDEBINFO:STRING=
CMAKE_SKIP_INSTALL_RPATH:BOOL=NO
CMAKE_SKIP_RPATH:BOOL=NO
CMAKE_STATIC_LINKER_FLAGS:STRING=
CMAKE_STATIC_LINKER_FLAGS_DEBUG:STRING=
CMAKE_STATIC_LINKER_FLAGS_MINSIZEREL:STRING=
CMAKE_STATIC_LINKER_FLAGS_RELEASE:STRING=
CMAKE_STATIC_LINKER_FLAGS_RELWITHDEBINFO:STRING=
CMAKE_STRIP:FILEPATH=/usr/bin/strip
CMAKE_VERBOSE_MAKEFILE:BOOL=FALSE
CPU_ENABLE:BOOL=ON
CUDA_ENABLE:BOOL=OFF
HWLOC_ENABLE:BOOL=OFF
MICROHTTPD_ENABLE:BOOL=OFF
OPENSSL_CRYPTO_LIBRARY:FILEPATH=/usr/lib/libcrypto.so
OPENSSL_INCLUDE_DIR:PATH=/usr/include
OPENSSL_SSL_LIBRARY:FILEPATH=/usr/lib/libssl.so
OpenCL_ENABLE:BOOL=OFF
OpenSSL_ENABLE:BOOL=ON
PKG_CONFIG_EXECUTABLE:FILEPATH=/usr/bin/pkg-config
XMR-STAK_COMPILE:STRING=native
  ```

I Get:

`
c:\Users\User\source>cmake --build . --config Release --target install
Microsoft (R) Build Engine version 15.4.8.50001 for .NET Framework
Copyright (C) Microsoft Corporation. All rights reserved.

Build started 26/03/2018 08:42:40.
Project "C:\Users\User\source\install.vcxproj" on node 1 (default targets).
Project "C:\Users\User\source\install.vcxproj" (1) is building "C:\Users\User\source\ZERO_CHECK.vcxproj" (2) on node 1
(default targets).
InitializeBuildStatus:
  Creating "x64\Release\ZERO_CHECK\ZERO_CHECK.tlog\unsuccessfulbuild" because "AlwaysCreate" was specified.
CustomBuild:
  All outputs are up-to-date.
FinalizeBuildStatus:
  Deleting file "x64\Release\ZERO_CHECK\ZERO_CHECK.tlog\unsuccessfulbuild".
  Touching "x64\Release\ZERO_CHECK\ZERO_CHECK.tlog\ZERO_CHECK.lastbuildstate".
Done Building Project "C:\Users\User\source\ZERO_CHECK.vcxproj" (default targets).

Project "C:\Users\User\source\install.vcxproj" (1) is building "C:\Users\User\source\ALL_BUILD.vcxproj" (3) on node 1 (
default targets).
Project "C:\Users\User\source\ALL_BUILD.vcxproj" (3) is building "C:\Users\User\source\xmr-stak.vcxproj" (4) on node 1
(default targets).
Project "C:\Users\User\source\xmr-stak.vcxproj" (4) is building "C:\Users\User\source\xmr-stak-backend.vcxproj" (5) on
node 1 (default targets).
Project "C:\Users\User\source\xmr-stak-backend.vcxproj" (5) is building "C:\Users\User\source\xmr-stak-c.vcxproj" (6) o
n node 1 (default targets).
InitializeBuildStatus:
  Creating "xmr-stak-c.dir\Release\xmr-stak-c.tlog\unsuccessfulbuild" because "AlwaysCreate" was specified.
CustomBuild:
  All outputs are up-to-date.
ClCompile:
  All outputs are up-to-date.
Lib:
  All outputs are up-to-date.
  xmr-stak-c.vcxproj -> C:\Users\User\source\bin\Release\xmr-stak-c.lib
FinalizeBuildStatus:
  Deleting file "xmr-stak-c.dir\Release\xmr-stak-c.tlog\unsuccessfulbuild".
  Touching "xmr-stak-c.dir\Release\xmr-stak-c.tlog\xmr-stak-c.lastbuildstate".
Done Building Project "C:\Users\User\source\xmr-stak-c.vcxproj" (default targets).

InitializeBuildStatus:
  Touching "xmr-stak-backend.dir\Release\xmr-stak-backend.tlog\unsuccessfulbuild".
CustomBuild:
  All outputs are up-to-date.
ClCompile:
  All outputs are up-to-date.
  C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.11.25503\bin\HostX64\x64\CL.exe /c /IC
  :\Users\User\source\. /I"C:\Program Files (x86)\AMD APP SDK\3.0\include" /I"C:\xmr-stak-dep\libmicrohttpd\include" /I
  "C:\xmr-stak-dep\openssl\include" /nologo /W3 /WX- /diagnostics:classic /O2 /Ob2 /D WIN32 /D _WINDOWS /D NDEBUG /D CO
  NF_NO_CUDA /D CONF_NO_HWLOC /D "BACKEND_TYPE=amd-cpu" /D _CRT_SECURE_NO_DEPRECATE /D NOMINMAX /D "CMAKE_INTDIR=\"Rele
  ase\"" /D _MBCS /Gm- /EHsc /MD /GS /fp:precise /Zc:wchar_t /Zc:forScope /Zc:inline /GR /Fo"xmr-stak-backend.dir\Relea
  se\\" /Fd"xmr-stak-backend.dir\Release\xmr-stak-backend.pdb" /Gd /TP /errorReport:queue C:\Users\User\source\xmrstak\
  backend\cpu\minethd.cpp
  minethd.cpp
c:\users\user\source\xmrstak\backend\cpu\crypto/cryptonight_aesni.h(432): warning C4244: 'initializing': conversion fro
m 'uint64_t' to 'uint8_t', possible loss of data [C:\Users\User\source\xmr-stak-backend.vcxproj]
C:\Users\User\source\.\xmrstak/misc/executor.hpp(67): warning C4244: 'initializing': conversion from 'double' to 'const
 ::size_t', possible loss of data [C:\Users\User\source\xmr-stak-backend.vcxproj]
c:\users\user\source\xmrstak\backend\cpu\autoAdjust.hpp(41): error C2039: 'IsCurrencyMonero': is not a member of 'jconf
' [C:\Users\User\source\xmr-stak-backend.vcxproj]
  C:\Users\User\source\.\xmrstak/jconf.hpp(11): note: see declaration of 'jconf'
c:\users\user\source\xmrstak\backend\cpu\autoAdjust.hpp(43): error C2065: 'MONERO_MEMORY': undeclared identifier [C:\Us
ers\User\source\xmr-stak-backend.vcxproj]
c:\users\user\source\xmrstak\backend\cpu\autoAdjust.hpp(48): error C2065: 'AEON_MEMORY': undeclared identifier [C:\User
s\User\source\xmr-stak-backend.vcxproj]
c:\users\user\source\xmrstak\backend\cpu\autoAdjust.hpp(107): warning C4267: '-=': conversion from 'size_t' to 'int32_t
', possible loss of data [C:\Users\User\source\xmr-stak-backend.vcxproj]
c:\users\user\source\xmrstak\backend\cpu\autoAdjust.hpp(109): warning C4267: '-=': conversion from 'size_t' to 'int32_t
', possible loss of data [C:\Users\User\source\xmr-stak-backend.vcxproj]
Done Building Project "C:\Users\User\source\xmr-stak-backend.vcxproj" (default targets) -- FAILED.

Done Building Project "C:\Users\User\source\xmr-stak.vcxproj" (default targets) -- FAILED.

Done Building Project "C:\Users\User\source\ALL_BUILD.vcxproj" (default targets) -- FAILED.

Done Building Project "C:\Users\User\source\install.vcxproj" (default targets) -- FAILED.


Build FAILED.

"C:\Users\User\source\install.vcxproj" (default target) (1) ->
"C:\Users\User\source\ALL_BUILD.vcxproj" (default target) (3) ->
"C:\Users\User\source\xmr-stak.vcxproj" (default target) (4) ->
"C:\Users\User\source\xmr-stak-backend.vcxproj" (default target) (5) ->
(ClCompile target) ->
  c:\users\user\source\xmrstak\backend\cpu\crypto/cryptonight_aesni.h(432): warning C4244: 'initializing': conversion f
rom 'uint64_t' to 'uint8_t', possible loss of data [C:\Users\User\source\xmr-stak-backend.vcxproj]
  C:\Users\User\source\.\xmrstak/misc/executor.hpp(67): warning C4244: 'initializing': conversion from 'double' to 'con
st ::size_t', possible loss of data [C:\Users\User\source\xmr-stak-backend.vcxproj]
  c:\users\user\source\xmrstak\backend\cpu\autoAdjust.hpp(107): warning C4267: '-=': conversion from 'size_t' to 'int32
_t', possible loss of data [C:\Users\User\source\xmr-stak-backend.vcxproj]
  c:\users\user\source\xmrstak\backend\cpu\autoAdjust.hpp(109): warning C4267: '-=': conversion from 'size_t' to 'int32
_t', possible loss of data [C:\Users\User\source\xmr-stak-backend.vcxproj]


"C:\Users\User\source\install.vcxproj" (default target) (1) ->
"C:\Users\User\source\ALL_BUILD.vcxproj" (default target) (3) ->
"C:\Users\User\source\xmr-stak.vcxproj" (default target) (4) ->
"C:\Users\User\source\xmr-stak-backend.vcxproj" (default target) (5) ->
(ClCompile target) ->
  c:\users\user\source\xmrstak\backend\cpu\autoAdjust.hpp(41): error C2039: 'IsCurrencyMonero': is not a member of 'jco
nf' [C:\Users\User\source\xmr-stak-backend.vcxproj]
  c:\users\user\source\xmrstak\backend\cpu\autoAdjust.hpp(43): error C2065: 'MONERO_MEMORY': undeclared identifier [C:\
Users\User\source\xmr-stak-backend.vcxproj]
  c:\users\user\source\xmrstak\backend\cpu\autoAdjust.hpp(48): error C2065: 'AEON_MEMORY': undeclared identifier [C:\Us
ers\User\source\xmr-stak-backend.vcxproj]

    4 Warning(s)
    3 Error(s)

Time Elapsed 00:00:03.18

c:\Users\User\source>`

--- a/.travis.yml
+++ b/.travis.yml
@@ -55,6 +55,22 @@ matrix:
         - CMAKE_C_COMPILER=gcc-6
         - XMRSTAK_CMAKE_FLAGS="-DCUDA_ENABLE=OFF -DOpenCL_ENABLE=OFF"
 
+    # test with disabled HWLOC, MICROHTTPD, OpenSSL and no accelerators
+    - os: linux
+      compiler: gcc
+      addons:
+        apt:
+          sources:
+            - ubuntu-toolchain-r-test
+          packages:
+            - *default_packages
+            - gcc-6
+            - g++-6
+      env:
+        - CMAKE_CXX_COMPILER=g++-6
+        - CMAKE_C_COMPILER=gcc-6
+        - XMRSTAK_CMAKE_FLAGS="-DCUDA_ENABLE=OFF -DOpenCL_ENABLE=OFF -DHWLOC_ENABLE=OFF -DOpenSSL_ENABLE=OFF -DMICROHTTPD_ENABLE=OFF"
+
     - os: linux
       compiler: gcc
       addons:

--- a/xmrstak/backend/cpu/autoAdjust.hpp
+++ b/xmrstak/backend/cpu/autoAdjust.hpp
@@ -35,19 +35,9 @@ public:
 
 	bool printConfig()
 	{
-		size_t hashMemSizeKB;
-		size_t halfHashMemSizeKB;
 
-		if(::jconf::inst()->IsCurrencyMonero())
-		{
-			hashMemSizeKB = MONERO_MEMORY / 1024u;
-			halfHashMemSizeKB = hashMemSizeKB / 2u;
-		}
-		else
-		{
-			hashMemSizeKB = AEON_MEMORY / 1024u;
-			halfHashMemSizeKB = hashMemSizeKB / 2u;
-		}
+		const size_t hashMemSizeKB = cn_select_memory(::jconf::inst()->GetMiningAlgo()) / 1024u;
+		const size_t halfHashMemSizeKB = hashMemSizeKB / 2u;
 
 		configEditor configTpl{};
 

